src/integrationTest/java/org/opensearch/security/CrossClusterSearchTests.java

```diff
@@ -155,14 +155,14 @@ public CrossClusterSearchTests(Boolean ccsMinimizeRoundtrips) {
 	@BeforeClass
 	public static void createTestData() {
 		try(Client client = remoteCluster.getInternalNodeClient()){
-			client.prepareIndex(SONG_INDEX_NAME).setId(SONG_ID_1R).setRefreshPolicy(IMMEDIATE).setSource(SONGS[0]).get();
-			client.prepareIndex(SONG_INDEX_NAME).setId(SONG_ID_6R).setRefreshPolicy(IMMEDIATE).setSource(SONGS[5]).get();
-			client.prepareIndex(PROHIBITED_SONG_INDEX_NAME).setId(SONG_ID_3R).setRefreshPolicy(IMMEDIATE).setSource(SONGS[1]).get();
-			client.prepareIndex(LIMITED_USER_INDEX_NAME).setId(SONG_ID_5R).setRefreshPolicy(IMMEDIATE).setSource(SONGS[4]).get();
+			client.prepareIndex(SONG_INDEX_NAME).setId(SONG_ID_1R).setRefreshPolicy(IMMEDIATE).setSource(SONGS[0].asMap()).get();
```

Comment:
out of curiosity, why this change? I see similar changed below too and am wondering if it has something to do with structure of query response.

Reply:
In order not to lose information about the types of fields that we have in the Song class, I changed the type of the Song.SONGS variable from Map[] to Song[]. I cannot pass song to the IndexRequestBuilder setSource(Object... source) method, it expects an array containing pairs of field name and value. I need to call song.asMap().